Class overhaul

glallen01 opened the pull request:
This includes some major breaking changes in order to undo early mistakes, but it makes the document much more latex like: (ie no `\Nii`'s ... just `\item`'s in `\enumerate`). There's still a few things I need to re-look, like spacing and optional items like suspenses, and the enclosures/copy lines are still a work in progress, but close as soon as I work in some conditionals. Since you and @pconwell were the major forks of this, please take a look at the update and consider a merge.

lassimus committed:
added Arial font install instructions to README

lassimus committed:
changed README to README.md

lassimus committed:
git rm old README

philoserf committed:
refactor: being pedantic

- clear all TeX lint (chktex && lacheck)
- reformat with latextindent
- use canonocal name for LICENSE file
- update GPL text
- use README.md and update formatting
- add gitignore file

no functional changes

glallen01 committed:
class overhaul

glallen01 committed:
class overhaul

glallen01 committed:
switch to lualatex, 3x runs to position classmark

glallen01 committed:
switch to lualatex, 3x runs to position classmark

glallen01 committed:
updated long example

glallen01 committed:
updated long example

glallen01 committed:
fixes memoline wrapping; hack for \AtEndDocument

glallen01 committed:
fixes memoline wrapping; hack for \AtEndDocument

glallen01 committed:
fixes from chktex, lacheck (thanks to @philoserf)

glallen01 committed:
fixes from chktex, lacheck (thanks to @philoserf)

glallen01 committed:
merged philoserf-contribs-being-pedantic

glallen01 committed:
Merge branch 'test' of github.com:glallen01/army-memorandum-class into test

glallen01 committed:
merges lassimus-master

glallen01 committed:
merges lassimus-master: Arial install cmds

glallen01 committed:
adds authline, fixes and comments

glallen01 committed:
updates README

glallen01 committed:
updates README

glallen01 committed:
switches from eps to pdf logo, adds examples pdfs

glallen01 committed:
update .gitignore

glallen01 committed:
removes boxes adds warnings for missing defs

glallen01 committed:
Updates readme

glallen01 committed:
added reply-to-attention-of

glallen01 committed:
fixes spacing #8, breaks #9 more

glallen01 committed:
add example to fouo marking; make marking parameter

glallen01 committed:
Fixes #14. re \authority

glallen01 committed:
makes \documentmark{} an optional arg for classification page marks

glallen01 committed:
adds optional \suspensedate{}

glallen01 committed:
fixes #9: end-matter overhaul

glallen01 committed:
fixes #19 - page # height

glallen01 committed:
messing with makefiles

glallen01 committed:
adds thru lines #16, but needs further review

glallen01 committed:
corrects footer spacing

glallen01 committed:
parameterizes address, logo; lots of other changes; fixes #15, #16

- logo not checked in, will back out in next commit.
- conflict between \pagedesign and 2nd line in \subject

pconwell commented:
Sorry, just getting around to looking at this. I'm running into an error with compiling. Looks like you've switched to lualatex in c38b279 but the ../example.tex still shows that the target compiler is pdflatex. This is (unsurprisingly) causing the following error:

```
c:/texlive/2018/texmf-dist/tex/latex/fontspec/fontspec.sty:45: Fatal Package fontspec Error: The fontspec package requires either XeTeX or
(fontspec)                      LuaTeX.
(fontspec)                      
(fontspec)                      You must change your typesetting engine to,
(fontspec)                      e.g., "xelatex" or "lualatex"instead of
(fontspec)                      "latex" or "pdflatex".
```

I'm assuming the switch is because of Arial font not being available in pdflatex (specifically that Arial is only available in fontspec which is not available in pdflatex). I started to see if I could do a workaround for users with pdflatex, something along the lines of:

``` latex
% Change the font if you want to, depending on whether you're using pdflatex or xelatex/lualatex
\ifxetexorluatex
  % If using xelatex or lualatex:
  \setmainfont{Arial}
\else
  % If using pdflatex:
  \usepackage[utf8]{inputenc}
  \usepackage[T1]{fontenc}
  \usepackage[default]{Helvet}
\fi
```

Which I know what helvet is technically different from Arial but it's 'close enough' that most users will never know the difference. I think giving the option between the two would be nice. If a user happens to already be using a pdflatex environment (which my understanding is pdflatex is the most popular on windows... but I could be wrong), they can either (1) install lualatex or (2) use a 'close enough' solution and keep using their existing environment.

However, before I could get past that, I started running into issues with reading the DODb1.pdf.

```
Syntax Warning: May not be a PDF file (continuing anyway)
Syntax Error: Couldn't read xref table
Syntax Warning: PDF file is damaged - attempting to reconstruct xref table...
Syntax Error: Couldn't find trailer dictionary
Syntax Error: Couldn't read xref table

!pdfTeX error: pdflatex.exe (file c:/Users/pconwell/Desktop/army-memorandum-class/examples/DODb1.pdf): xpdf: reading PDF image failed
 ==> Fatal error occurred, no output PDF file produced!
Collected error summary (may duplicate other messages):
  pdflatex: Command for 'pdflatex' gave return code 1
      Refer to 'c:/Users/pconwell/Desktop/army-memorandum-class/examples/example.log' for details
=== TeX engine is 'pdfTeX'
Latexmk: Errors, so I did not complete making targets
```

I'll keep looking at it, don't have a lot of time at work. The DODb1.pdf file seems to be fine, so I'm not sure why it's throwing an error. My guess (without looking) is luatex is happy importing a pdf, but pdflatex is not. But - again - I haven't really looked at it so I'm not sure.

nelsonrg commented:
This looks like a serious improvement over the previous approach. Because of the large quantity of changes, I've put the pull request into its own branch for testing. I'll test and modify it before I merge.

Commit:
Remove reply to attention of, unslant header text, remove signature date from headers after page 1, remove Encl if there are no enclosures

glallen01 committed:
Merge pull request #25 from stevewillson/master

Remove reply to attention of, unslant header text, remove signature d…

glallen01 committed:
updated examples from last merge

glallen01 committed:
updates README.md and examples/example.tex with /address{}

nstickney committed:
automatically format \today dates

nstickney committed:
automatically uppercase signature name

glallen01 committed:
Merge pull request #32 from usma-eecs/upstream-fixes

Uppercase author, and automatically format date

glallen01 committed:
AR 25-50 1.7a -- suspense line should be bold

glallen01 committed:
no extra space after periods, AR 25-50, 1-39.a

glallen01 committed:
fix makefile, use latexrun if available, otherwise latexmk

glallen01 committed:
Merge pull request #38 from glallen01/ar25-50-update

Ar25 50 update

jschless committed:
multifor and enclosure bug fix

glallen01 committed:
Merge branch 'jschless-master'

glallen01 committed:
Fixes #40

A difference between TL2022 and TL2023 affected the spacing and indentation of
the first item ending after `\dodheader`, so swithching `\dodheader` to ending
with a `\par` and adjusting the `baselineskip` seemed to fix this with TL2023
and still work with 2022.

glallen01 committed:
Update armymemo.cls

bump minor version for #40 fix

glallen01 committed:
Fixes #42

glallen01 committed:
Merge pull request #43 from glallen01/address-fix

Fixes #42

glallen01 committed:
Add note on font re #5

glallen01 committed:
Cherry-pick AR compliance fixes from PR #12

- \brokenpenalty=10000: prevent hyphenation across page breaks (AR 25-50 2-5-c-3)
- SUBJECT:~~ double space after colon (AR 25-50 1-39.b)
- \enclsnocount: new command for unnumbered enclosure list (AR 25-50 Figure 4-4)
- Refactor \am@encls to proper if/else chain

glallen01 committed:
feat: bundle digsig.sty for optional PDF signature fields

Adds Martin Lottermoser's digsig.sty (v2.3, MIT license) to enable
PDF digital signature field support via hyperref. The package is bundled
because it's not available in TeX Live. A symlink in examples/ allows
compilation without TEXINPUTS configuration.

Source: https://gitlab.mn.tu-dresden.de/nsm/templates/nsm-proposal/-/raw/93a60a51e98b07ed4de3480a5ce1cb034f3e5c86/digsig.sty

glallen01 committed:
feat(digsig): add optional [digsig] class option for PDF signature fields

- Add [digsig] class option backed by etoolbox boolean am@digsig
- Conditionally load bundled digsig.sty (v2.3, MIT) after hyperref
- Render 3in x 3\baselineskip signature field in TikZ overlay above signer name
- Add examples/example-sig.tex demonstrating the option with a custom logo
- Add CHANGES.md [Unreleased] section summarizing branch changes

glallen01 committed:
update example pdfs

glallen01 committed:
Merge branch 'update-2026-03-28' into digsig-2026-03-28

glallen01 committed:
Merge pull request #46 from glallen01/digsig-2026-03-28

feat(digsig): add [digsig] class option for PDF digital signature fields

glallen01 committed:
update changes

glallen01 committed:
Merge pull request #45 from glallen01/update-2026-03-28

AR compliance fixes: brokenpenalty, SUBJECT spacing, enclsnocount
Add optional digital signature box

glallen01 committed:
restores Times; bumps version

glallen01 committed:
Merge pull request #47 from glallen01/update-version

restores Times New Roman as default
bumps version
